engine/access/rpc/backend/backend_stream_transactions.go

```diff
@@ -30,55 +33,141 @@ type backendSubscribeTransactions struct {
+// If the block ID cannot be determined or an error occurs during subscription creation, a failed subscription is returned.
+func (b *backendSubscribeTransactions) SubscribeTransactionStatusesFromStartBlockID(
+	ctx context.Context,
+	txID flow.Identifier,
+	startBlockID flow.Identifier,
+	requiredEventEncodingVersion entities.EventEncodingVersion,
+) subscription.Subscription {
+	return b.createSubscription(ctx, txID, startBlockID, 0, flow.ZeroID, requiredEventEncodingVersion, false)
+}
+
+// SubscribeTransactionStatusesFromLatest subscribes to the status updates of a transaction.
+// Monitoring starts from the latest block.
+// If the block cannot be retrieved or an error occurs during subscription creation, a failed subscription is returned.
+func (b *backendSubscribeTransactions) SubscribeTransactionStatusesFromLatest(
+	ctx context.Context,
+	txID flow.Identifier,
+	requiredEventEncodingVersion entities.EventEncodingVersion,
+) subscription.Subscription {
+	header, err := b.txLocalDataProvider.state.Sealed().Head()
 	if err != nil {
-		return subscription.NewFailedSubscription(err, "could not get start height")
+		b.log.Error().Err(err).Msg("failed to retrieve latest block")
+		return subscription.NewFailedSubscription(err, "failed to retrieve latest block")
 	}
 
-	txInfo := TransactionSubscriptionMetadata{
+	return b.createSubscription(ctx, txID, header.ID(), 0, flow.ZeroID, requiredEventEncodingVersion, false)
+}
+
+// createSubscription initializes a subscription for monitoring a transaction's status.
+// If the start height cannot be determined, a failed subscription is returned.
+func (b *backendSubscribeTransactions) createSubscription(
+	ctx context.Context,
+	txID flow.Identifier,
+	startBlockID flow.Identifier,
+	startBlockHeight uint64,
+	referenceBlockID flow.Identifier,
+	requiredEventEncodingVersion entities.EventEncodingVersion,
+	shouldTriggerPending bool,
+) subscription.Subscription {
+	var nextHeight uint64
+	var err error
+
+	// Get height to start subscription from
+	if startBlockID == flow.ZeroID {
+		if nextHeight, err = b.blockTracker.GetStartHeightFromHeight(startBlockHeight); err != nil {
+			b.log.Error().Err(err).Uint64("block_height", startBlockHeight).Msg("failed to get start height")
+			return subscription.NewFailedSubscription(err, "failed to get start height")
+		}
+	} else {
+		if nextHeight, err = b.blockTracker.GetStartHeightFromBlockID(startBlockID); err != nil {
+			b.log.Error().Err(err).Str("block_id", startBlockID.String()).Msg("failed to get start height")
+			return subscription.NewFailedSubscription(err, "failed to get start height")
+		}
+	}
+
+	// choose initial transaction status
+	initialStatus := flow.TransactionStatusUnknown
+	if shouldTriggerPending {
+		// The status of the first pending transaction should be returned immediately, as the transaction has already been sent.
+		// This should occur only once for each subscription.
+		initialStatus = flow.TransactionStatusPending
```

Comment:
since there is already special handling for pending, you could move this to `handlePendingStatus` and keep the logic here simpler